### ISSUE TITLE
keras.models.Model should not require either inputs nor outputs?

### ISSUE THREAD

**Issue by cottrell** [keras-team/keras#12630]
Am trying to do some testing of generic optimization capabilities. It feels like there is no reason a graph could not be optimized via a keras model that takes no inputs or outputs but simply takes a loss function and optimizer. 

I ended up in this rabbit hole trying out tf 2.0 and hitting issues doing the same there. Would be good to understand what is the intended workflow between the two in 2.0 for vanilla optimizations.
